Scripts/RemediationScripts/Remediate-DisableLocalAccountsForKubernetesService.ps1

```diff
@@ -0,0 +1,867 @@
+        }
+        catch 
+        {
+            Write-Host "Error occurred while fetching auth header. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)   
+        }
+
+        return($headers)
+    }
+
+
+    [PSObject] GetKuberneteCluster([string] $subscriptionId)
+    {
+        $content = $null
+        $kubernetesClusters = @()
+        try
+        {
+            $armUri = 
+            "https://management.azure.com/subscriptions/$($subscriptionId)/providers/Microsoft.ContainerService/managedClusters?api-version=2023-01-01"
+            $headers = $this.GetAuthHeader()
+            $method = "GET"
+            # API to set local accounts Profile config to Kubernetes cluster
+            $response = Invoke-WebRequest -Method $method -Uri $armUri -Headers $headers -UseBasicParsing
+            $content = ConvertFrom-Json $response.Content
+
+            $content.value | ForEach-Object {
+                    $resourceId = $_.id
+                    $name = $_.name
+                    $location = $_.location
+                    $enableRBAC = $_.properties.enableRBAC
+                    $aadProfile = $_.properties.aadProfile
+                    $disableLocalAccounts = $_.properties.disableLocalAccounts  
+                    $kubernetesClusters += $_ | Select-Object   @{N='Id';E={$resourceId}},
+                                                                @{N='Name';E={$name}},
+                                                                @{N='Location';E={$location}},
+                                                                @{N='EnableRBAC';E={$enableRBAC}},
+                                                                @{N='AadProfile';E={$aadProfile}},
+                                                                @{N='DisableLocalAccounts';E={$disableLocalAccounts}} 
+ 
+            }
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting local accounts profile to Kubernetes Cluster. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+        }
+        
+        return($kubernetesClusters)
+    }
+
+    [PSObject] DisableLocalAccountsForKuberneteCluster([string] $subscriptionId, [string] $resourceName, [string] $resourceGroup, [string] $location, [boolean] $isAADEnabled, [Object[]] $aadClientIds)
+    {
+        $content = $null
+        try
+        {
+            $armUri = "https://management.azure.com/subscriptions/$($subscriptionId)/resourceGroups/$($resourceGroup)/providers/Microsoft.ContainerService/managedClusters/$($resourceName)?api-version=2022-02-01"
+            $headers = $this.GetAuthHeader()
+            $method = "Put"
+            
+            if($isAADEnabled)
+            {
+                $body =@'
+	            {
+                    "location": "{0}",
+                    "properties": {
+                        "disableLocalAccounts":true
```

Comment:
we dont have PS command to do these changes?

Reply:
No. Richa in previous script for kubernetes we followed the same approach. In my case i need to make the API call to even fetch the property i require.